### PR TITLE
[Serializer] snake properties loose values

### LINKED ISSUE #36288: [Serializer] Groups not working
**Symfony version(s) affected**: 4.4.x

**Description**  
Hi, 

I'm using groups to serialize data but it's not always working.

For example : 

```
    /**
     * @ORM\Column(type="array", nullable=true)
     * @Groups({"me_user", "write_user"})
     */
    protected $register_purposes;
```

```
        //return serialized data
        $serializedEntity = $this->container->get('serializer')->serialize($users, 'json', array('groups' => ['me_user']));
        return new Response($serializedEntity, 200, ['Content-Type' => 'application/json']);
```

Won't work until I passed the field with public access. 
So : public $register_purposes;

Any idea why ?

### PR DESCRIPTION
**Pull request by lmasforne**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #36288 
| License       | MIT

Fix bug when a snake property value is lost in serialization. 
We try to guess name method on GetSetMethodNormalizer but not we have the good name 'getSnake_case' when in general it's 'getSnakeCase'

### CODE REVIEW

File: src/Symfony/Component/Serializer/Normalizer/GetSetMethodNormalizer.php [hunk 1]
```diff
@@ -11,15 +11,13 @@
 
 namespace Symfony\Component\Serializer\Normalizer;
 
-/**
+use Symfony\Component\Serializer\NameConverter\CamelCaseToSnakeCaseNameConverter; /**
```
Comment (Kocal) [inline src/Symfony/Component/Serializer/Normalizer/GetSetMethodNormalizer.php:14]:
```suggestion
use Symfony\Component\Serializer\NameConverter\CamelCaseToSnakeCaseNameConverter;

/**
```

File: src/Symfony/Component/Serializer/Tests/Fixtures/GroupDummySnakeCase.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
+<?php
+
```
Comment (Kocal) [inline src/Symfony/Component/Serializer/Tests/Fixtures/GroupDummySnakeCase.php:2]: The license header is missing

Comment (lmasforne) [reply src/Symfony/Component/Serializer/Tests/Fixtures/GroupDummySnakeCase.php:2]: Yes sorry for that, i begin i try to dont forget next time ;) . thanks

Comment (Kocal) [reply src/Symfony/Component/Serializer/Tests/Fixtures/GroupDummySnakeCase.php:2]: No proble :)

**Comment by dunglas** [comment]
👎 on my side, because:

1. Symfony doesn't want to support all possible combinations of conventions. The supported conventions are documented and known for years. Mixing snake case (for property names) and camel case (for method names) isn't supported.
2. Finding the property to access in a smart way is the responsibility of PropertyAccess, not of Serializer
3. It's already possible to support such conventions userland by using the `ObjectNormalizer` along with a custom implementation of the `PropertyAccessorInterface`

**Comment by lmasforne** [comment]
@dunglas ok, i understand, thanks for your feedbacks,
I just put some links to help the person who reported the bug.

**Pull request closed without merging**